ee/allowedcmd/cmd.go

```diff
@@ -13,17 +13,55 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+
+	"github.com/kolide/launcher/pkg/traces"
 )
 
-type AllowedCommand func(ctx context.Context, arg ...string) (*exec.Cmd, error)
+type AllowedCommand func(ctx context.Context, arg ...string) (*TracedCmd, error)
+
+type TracedCmd struct {
+	*exec.Cmd
+}
+
+// Start overrides the Start method to add tracing before executing the command.
+func (t *TracedCmd) Start() error {
+	_, span := traces.StartSpan(context.Background(), "path", t.Cmd.Path)
```

Comment:
maybe?

```suggestion
	_, span := traces.StartSpan(context.Background(), "path", t.Cmd.Path, "args", t.Cmd.Args)
```

Reply:
I think would want to `fmt.Sprintf("%+v", t.Cmd.Args)` -- our trace frontend doesn't allow for array attributes unfortunately